pkg/cortex/client/cortex/client.py

```diff
@@ -88,6 +95,27 @@ def create_api(
                 yaml.dump([api_spec], f)  # write a list
                 return self._deploy(cortex_yaml_path, force, wait)
 
+        api_kind = api_spec.get("kind")
+        if api_kind == "TrafficSplitter":
+            if predictor:
+                raise ValueError(f"`predictor` parameter cannot be specified for {api_kind} kind")
+            if task:
+                raise ValueError(f"`task` parameter cannot be specified for {api_kind} kind")
+        elif api_kind == "TaskAPI":
+            if predictor:
+                raise ValueError(f"`predictor` parameter cannnot be specified for {api_kind} kind")
+            if task is None:
+                raise ValueError(f"`task` parameter must be specified for {api_kind} kind")
+        elif api_kind in ["BatchAPI", "RealtimeAPI"]:
+            if not predictor:
+                raise ValueError(f"`predictor` parameter cannot be specified for {api_kind}")
+            if task:
+                raise ValueError(f"`task` parameter must be specified for {api_kind}")
```

Comment:
nit: error message needs to be updated to `task parameter must not be specified for {api_kind}`